### ISSUE TITLE
[TF2] Remove sv_cheats requirement from tf_bot_spawn_use_preset_roster and tf_bot_reevaluate_class_in_spawnroom

### ISSUE THREAD

**Issue by Whurrhurr** [ValveSoftware/Source-1-Games#4469]
title, some players (like myself) liked playing with tfbots that weren't using preset class rosters and didn't randomly suicide in spawn just to change classes. i get this was supposedly a thing for the official competitive mode, but since matches don't count if someone disconnects at any point, the only time you'd ever see a TFBot playing in valve comp is if the matchmaker couldn't find 12 players for a match.

Why i think the pre-set class rosters for TFBots is bad:
 - You will face the exact same team composition from a team of TFBots every time. the only time it changes is depending on what team they're on in a Attack/Defend or Payload gamemodes, if their team owns the control point in the King of The Hill gamemode, other gamemodes use the same static preset roster for both teams, meaning both teams will have the exact same team composition every round.
 - They suicide when changing class if they spawn and decide to play a different class. and according to the description of tf_bot_reevaluate_class_in_spawnroom:  bots will opportunisticly switch class while in spawnrooms if their current class is no longer their first choice. (what does this even mean?) with the console command set to 0, tfbots will just respawn as a different class. which is just the same thing, but without them suiciding in spawn to change classes.
 - Certain classes won't show up unless a team has more than X amount of players. (e.g. no TFBot will pick sniper unless their team has 10 players at the minimum, and no TFBot will pick spy unless they have a full team of 12.)
 - The preset roster only goes up to 12 players, not the maximum 16 that you can have with an increased maxplayers amount. so you can no longer host your own listenserver with a 16v16 with 31 TFBots, because 7 of those bots will pack "undefined" and never spawn.

**Comment by HeteroChromia42** [comment]
Agreed. This is also particularly more relevant now with the possibility of 100 Player Slots. Sure, that's locked behind a launch variable, but if i want to utterly melt my CPU and have a epic 50v50 bot battle, i should be able to.

**Comment by amppiapina** [comment]
agreed
